tests/ashley/test_templatetags_custom_tags.py

```diff
@@ -0,0 +1,265 @@
+        topic = TopicFactory(forum=forum, poster=user1, subject="topic création")
+
+        # Chek that user1 is not instructor
+        assert get_rendered(topic, user1) == "NO"
+        # Chek that user2 is instructor
+        assert get_rendered(topic, user2) == "YES"
+
+        # Add forum to context2 where user1 has role "instructor"
+        forum.lti_contexts.add(context2)
+        # Check that user1 is now instructor as well
+        assert get_rendered(topic, user1) == "YES"
+
+    def test_render_block_is_instructor_of_this_forum(self):
+        """
+        check that the instructor icon is present on the different views
+        as expected
+        """
+        created_by = 'By: <a href="{{url}}">{{ username }}</a>'
+        icon = '<i class="icon_writer fas fa-award" aria-hidden="true" title="Instructor"></i>'
+
+        def get_rendered(topic, user, selector=""):
+            """render template provided"""
+            template = Template(selector + created_by + icon)
+            context = Context(
+                {
+                    "topic": topic,
+                    "user": user,
+                    "username": get_forum_member_display_name(user),
+                    "url": f"/forum/member/profile/{user.id}/",
+                }
+            )
+            rendered = template.render(context)
+
+            return self._trim(rendered)
+
+        def get_rendered_topic_by(topic, user):
+            """expected rendered where topic has been created by """
+            selector = 'topic-created">'
+            return get_rendered(topic, user, selector)
+
+        def get_rendered_post_by(topic, user):
+            """expected rendered from post has been created by """
+            selector = 'topic-last-post">'
+            return get_rendered(topic, user, selector)
+
+        # Create two users
+        user1 = UserFactory(public_username="instructor")
+        user2 = UserFactory(public_username="student")
+        # Create an LTI Context
+        context = LTIContextFactory()
+        # Sync user1 groups in context with role "instructor"
+        context.sync_user_groups(user1, ["instructor"])
+        # Sync user1 groups in context with role "student"
+        context.sync_user_groups(user2, ["student"])
+
+        # Create forum and add context
+        forum = ForumFactory(name="forum_test")
+        forum.lti_contexts.add(context)
+
+        # assign permission to the group for this forum
+        self._init_forum(forum, context)
+
+        # Set up topic
+        topic1 = TopicFactory(forum=forum, poster=user1, subject="topic by instructor")
+
+        # Set up post
+        PostFactory.create(
+            topic=topic1,
+            poster=user1,
+            subject="post by instructor",
+        )
+
+        self.client.force_login(user1)
+
+        # accessing forum view
+        response = self.client.get(reverse("forum:index"))
+        # check that it contains the icon instructor
+        self.assertIn(
```

Comment:
you should be able to use
```
self.assertContains(response, rendered_content, html=True) 
```
Which takes care of trimming and utf8 encoding.

Reply:
ah ah ah...

Reply:
Thank you to have share this option with me, I hadn't seen it yesterday. I'm sure in the futur it will be of good use.
Unfortunately I've tried again with this solution and I can't manage to find what I'm looking for, I think my problem remains on "\n". As my previous solution is working and as I've already spent a good bit of time on it yesterday, I submit again with no change on this point.
```
        By: <a href="/forum/member/profile/3/">instructor</a>




    <i class="icon_writer fas fa-award" aria-hidden="true" title="Instructor"></i>
```